src/lib/form_action.ts

```diff
@@ -10,15 +10,15 @@ type FormActionMessage = {
 
 export const form_action = (
 	opts?: FormActionMessage,
-	pre?: (data?: any | unknown) => any,
+	pre?: (data?: unknown) => unknown,
 	callback?: (
-		data?: any | unknown,
+		data?: unknown,
```

Comment:
This should really be a generic so we can infer the returned data type - having it typed as unknown here just makes every form action now have a type error. I think it can be inferred from the from action?

Reply:
Good question about the inferring from action. I'll have to take a look because at a glance not sure how.